src/components/text.components.js

```diff
@@ -0,0 +1,31 @@
+import $ from 'jquery';
+import Component from './component';
+
+export default class TextInput extends Component {
+    constructor(stringQuery, _container, required = true, disabled = false) {
+        super(_container, disabled, required);
+        this.stringQuery = stringQuery;
+
+        this._container = $(`${stringQuery} input[type="text"]`);
+        this._textValid = false;
+    }
+
+    setRequired(boolean) {
+        this._container.prop('required', boolean);
+        this.required = boolean;
+    }
+
+    setInput(value) {
```

Comment:
The name does not correspond to the behavior. You are not setting an input, but a value.

Reply:
changed to setValue